protocol/testing/containertest/testnet_test.go

```diff
@@ -291,7 +290,54 @@ func TestUpgrade(t *testing.T) {
 	require.NoError(t, err)
 	defer testnet.MustCleanUp()
 	node := testnet.Nodes["alice"]
+	nodeAddress := constants.AliceAccAddress.String()
+	err = upgradeTestnet(nodeAddress, t, node)
+	require.NoError(t, err)
+}
+
+func TestStateUpgradeForPerpetualMarketType(t *testing.T) {
```

Comment:
I wouldn't make this test specific to perpetual market type but rather something that encompasses all state changes from one version to another.
I think it might be helpful to just call this TestStateUpgrade and move it into the uprade 5.0.0 folder if thats possible

Reply:
Good point! made the change so that its easier to add other state change tests. 

With regards to move it to upgrade 5.0.0 folder - isnt that for unit tests? all container tests are in the current folder

Reply:
I don't think it needs to be just for unit tests. It'll be nice to have this in the 5.0.0 folder because the next time someone is doing this for the next upgrade, they can just copy paste the 5.0.0 folder and have the entire basic framework for the upgrade changes

@teddyding to confirm if we are able to move this to the 5.0.0 folder

Reply:
Yeah agree with @shrenujb - we don't strictly differentiate tests based on file paths, but instead by functionality. Most custom modules (e.g. `x/clob`) have both unit and e2e tests within their directories. So it makes sense to all testing related to `v5.0.0` under `protocol/app/upgrades/v5.0.0`. 

We can highlight the test type by naming the file something like `upgrade_container_test` under `protocol/app/upgrades/v5.0.0`

Reply:
Since the function depends significantly on container tests, moving the function over and importing references causes a circular dependency. Maybe creating another upgrade test folder within `protocol/testing` would be a better idea?